Void/Void.ViewModel/VoidController.fs

```diff
@@ -39,67 +43,86 @@ type ViewController
+        | Event.LastWindowClosed ->
+            _view.Close()
+        | Event.EditorInitialized editor ->
+            _viewModel <- ViewModel.loadBuffer (Editor.currentBuffer editor) _viewModel 
+            // TODO duplication of BufferLoadedIntoWindow below
+            _viewModel.VisibleWindows.[0].Buffer
+            |> Render.bufferAsDrawingObjects _convert (ViewModel.wholeArea _viewModel)
             |> x.bufferDrawings
-            Command.Redraw
-        | Event.CoreInitialized -> x.initializeView()
-        | _ -> Command.Noop
+            _convert.cellBlockToPixels (ViewModel.wholeArea _viewModel) |> _view.TriggerDraw // TODO shouldn't redraw the whole UI
+        | _ -> ()
+        Command.Noop
 
-type MainController
+type Broker
     (
-        _view : MainView
+        _commandHandlers : (Command -> Command) list,
+        _eventHandlers : (Event -> Command) list,
+        _viewCtrl : ViewController,
+        _normalCtrl : NormalModeController
     ) =
-    let _normalCtrl = NormalModeController()
-    let _coreCtrl = CoreController()
-    let _viewCtrl = ViewController(_view)
-    let _eventHandlers = [_coreCtrl.handleEvent; _viewCtrl.handleEvent]
-
-    member x.initializeVoid() =
-        x.handleCommand <| Command.PublishEvent Event.CoreInitialized
-        x.handleCommand Command.ViewTestBuffer // for debugging
 
-    member x.handleViewEvent viewEvent =
+    member x.brokerViewEvent viewEvent =
         match viewEvent with
         | ViewEvent.PaintInitiated draw ->
             _viewCtrl.paint draw
         | ViewEvent.KeyPressed keyPress ->
-            _normalCtrl.handle keyPress |> x.handleCommand
+            _normalCtrl.handleKeyPress keyPress |> x.brokerCommand
         | ViewEvent.TextEntered text ->
             () // TODO implement input and command modes, etc
 
-    member private x.handleCommand command =
-        let notImplemented() =
-            Event.ErrorOccurred Error.NotImplemented
-            |> Command.PublishEvent
-            |> x.handleCommand
-
+    member x.brokerCommand command =
         match command with
-        | Command.ChangeToMode _
-        | Command.Edit
-        | Command.Yank
-        | Command.Put
-        | Command.FormatCurrentLine ->
-            notImplemented()
         | Command.PublishEvent event ->
             for handle in _eventHandlers do
-                handle event |> x.handleCommand
-        | Command.Quit
-        | Command.Redraw ->
-            _viewCtrl.handleCommand command |> x.handleCommand
-        | Command.ViewTestBuffer -> 
-            _coreCtrl.handleCommand command |> x.handleCommand
+                handle event |> x.brokerCommand
         | Command.Noop -> ()
+        | _ ->
+            for handle in _commandHandlers do
+                handle command |> x.brokerCommand
+
+module Init =
+    let initializeVoid view =

```

Comment:
Inspired by a comment you made that it didn't make sense to have it in the `VoidController`. :)